### PR TITLE
adding repayment to downpayment calc

### PR DESCRIPTION
**Pull request by cheedinmar**
Work in progress

### CODE REVIEW

File: src/components/AutoComplete.vue [hunk 1]
```diff
@@ -35,32 +35,36 @@ export default {
...
         selectItem(data) {
             this.inputValue = this.formatInput(data);
             this.apiLoaded = false;
             this.$emit("childToParent", data);
+            
         },
         searchEvent() {
             this.getproduct();
+
+            // this.inputValue = this.currentValue;
```
Comment (teebarg) [inline src/components/AutoComplete.vue:67]: remove comment

File: src/components/PaymentLog.vue [hunk 2]
```diff
@@ -313,16 +315,40 @@
...
+                  class="col d-flex align-items-center"
+                  :class="this.customer.guarantor_paystack.length > 0 ?'': 'disableToggle' "
+                  v-if="(addDownpayment && isAltaraPay) || stillShowToggle"
                 >
-                  View Amortization
-                </button>
-                <br />
+                  <toggle-button
+                    v-on:valueChangedEvent="triggerToggleEvent"
+                    :switchName="'addDownpayment'"
```
Comment (teebarg) [inline src/components/PaymentLog.vue:328]: Just use `switchName="addDownpayment"`

File: src/components/PaymentLog.vue [hunk 3]
```diff
@@ -313,16 +315,40 @@
...
+                    :key="'addDownpayment'"
```
Comment (teebarg) [inline src/components/PaymentLog.vue:329]: Just use `key="addDownpayment"`

File: src/components/PaymentLog.vue [hunk 4]
```diff
@@ -313,16 +315,40 @@
...
+                    :defaultState="addDownpayment"
+                    :label="'Add Repayment'"
```
Comment (teebarg) [inline src/components/PaymentLog.vue:331]: Just use `label="Add Repayment"`

File: src/components/PaymentLog.vue [hunk 5]
```diff
@@ -691,6 +762,29 @@ export default {
...
+        return firstvalue;
+      } else return secondvalue;
+    },
+    watchCashPrice() {
+
+      if (
+        this.salesLogForm?.product?.product?.category == "cash loan" &&
+        this.salesLogForm?.repayment_duration_id?.name == "six_months"
+      ) {
+        this.addDownpayment =
```
Comment (teebarg) [inline src/components/PaymentLog.vue:776]: Can we add a comment explaining what is happening here?

File: src/components/PaymentLog.vue [hunk 6]
```diff
@@ -936,6 +1038,30 @@ export default {
         this.rPayment = rePayment;
         this.pPrice = total;
         this.test1 = false;
+        const months = this.rDuration / 30;
+
+        const cycle = Math.ceil(28 / this.repaymentCircle);
+        const additionalRepayment = this.rPayment / (months * cycle);
+
+        if (
+          (this.selectedProduct.price > 80000 &&
```
Comment (teebarg) [inline src/components/PaymentLog.vue:1047]: Isn't this same logic we have here `watchCashPrice`?

File: src/components/PaymentLog.vue [hunk 7]
```diff
@@ -1104,13 +1232,16 @@ export default {
       this.serial === true ? (this.serial = false) : (this.serial = true);
     },
     toggleProductType() {
+      this.test0=true
```
Comment (teebarg) [inline src/components/PaymentLog.vue:1235]: test0? let's use a more descriptive name

File: src/components/PaymentLog.vue (hunk 6 above)
Comment (cheedinmar) [reply src/components/PaymentLog.vue:1047]: no the watchCashPrice is checking whether to display additional downpayment or not, this logic here is to determine how much repayment to be added, whether one or two or half,

**Review by teebarg — APPROVED** [review]

**Pull request merged**